File: Frontend-BAP/components/SimilarProducts.js
import React from 'react';
import {
  View,
  Text,
  FlatList,
  Image,
  TouchableOpacity,
} from 'react-native';
import tw from 'tailwind-react-native-classnames';

const SimilarProducts = ({ relatedItems, navigation }) => {
  if (!relatedItems || relatedItems.length === 0) {
    return null;
  }

  const renderProductCard = ({ item }) => {
    const providerName = item.providerName || item.provider?.descriptor?.name || 'Unknown Seller';
    const imageUrl = item.image || item.descriptor?.images?.[0] || 'https://via.placeholder.com/150';
    const priceValue = item.price?.value || '0.00';
    const currency = item.price?.currency || 'INR';
    const unit = item.quantity?.unitized?.measure?.unit || '';

    return (
      <TouchableOpacity
        onPress={() =>
          navigation.navigate('ProductDetails', { item_id: item.id })
        }
        style={tw`w-44 mr-4 bg-green-50 rounded-xl p-4 shadow-md border border-green-100`}
      >
        {/* Product Image */}
        <Image
          source={{ uri: imageUrl }}
          style={tw`w-full h-36 rounded-lg mb-3`}
          resizeMode="contain"
        />

        {/* Product Name */}
        <Text
          style={tw`text-sm font-semibold text-gray-800 mb-1`}
          numberOfLines={2}
        >
          {item.descriptor?.name || 'Unnamed Product'}
        </Text>

        {/* Price */}
        <Text style={tw`text-green-600 font-bold text-sm mb-1`}>
          ₹{priceValue} / {unit}
        </Text>

        {/* Seller Info */}
        <Text style={tw`text-green-700 text-xs font-medium`} numberOfLines={1}>
          Seller: {providerName}
        </Text>
      </TouchableOpacity>
    );
  };

  return (
    <View style={tw`mt-8`}>
      <Text style={tw`text-xl font-bold px-2 mb-3 text-gray-800`}>
        Similar Products
      </Text>
      <FlatList
        horizontal
        data={relatedItems}
        keyExtractor={(item) => item.id}
        renderItem={renderProductCard}
        showsHorizontalScrollIndicator={false}
        contentContainerStyle={tw`px-2`}
      />
    </View>
  );
};

export default SimilarProducts;

File: demo/lib/components/similar_products.dart
import 'package:flutter/material.dart';

class SimilarProducts extends StatelessWidget {
  final List<dynamic> relatedItems;
  final Function(String) onProductTap;

  const SimilarProducts({
    Key? key,
    required this.relatedItems,
    required this.onProductTap,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    if (relatedItems.isEmpty) return const SizedBox.shrink();

    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        const Padding(
          padding: EdgeInsets.symmetric(horizontal: 8.0, vertical: 12),
          child: Text(
            "Similar Products",
            style: TextStyle(
              fontSize: 20,
              fontWeight: FontWeight.bold,
              color: Colors.black87,
            ),
          ),
        ),
        SizedBox(
          height: 260, // Height for product card list
          child: ListView.builder(
            scrollDirection: Axis.horizontal,
            padding: const EdgeInsets.symmetric(horizontal: 8),
            itemCount: relatedItems.length,
            itemBuilder: (context, index) {
              final item = relatedItems[index];

              final providerName = item["providerName"] ??
                  item["provider"]?["descriptor"]?["name"] ??
                  "Unknown Seller";

              final imageUrl = item["image"] ??
                  (item["descriptor"]?["images"] != null &&
                          (item["descriptor"]["images"] as List).isNotEmpty
                      ? item["descriptor"]["images"][0]
                      : "https://via.placeholder.com/150");

              final priceValue = item["price"]?["value"] ?? "0.00";
              final currency = item["price"]?["currency"] ?? "INR";
              final unit =
                  item["quantity"]?["unitized"]?["measure"]?["unit"] ?? "";

              return GestureDetector(
                onTap: () => onProductTap(item["id"]),
                child: Container(
                  width: 170,
                  margin: const EdgeInsets.only(right: 12),
                  padding: const EdgeInsets.all(12),
                  decoration: BoxDecoration(
                    color: Colors.green.shade50,
                    borderRadius: BorderRadius.circular(16),
                    border: Border.all(color: Colors.green.shade100),
                    boxShadow: [
                      BoxShadow(
                        color: Colors.black.withOpacity(0.05),
                        blurRadius: 4,
                        offset: const Offset(2, 2),
                      ),
                    ],
                  ),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      // Product Image
                      ClipRRect(
                        borderRadius: BorderRadius.circular(12),
                        child: Image.network(
                          imageUrl,
                          height: 120,
                          width: double.infinity,
                          fit: BoxFit.contain,
                        ),
                      ),
                      const SizedBox(height: 10),

                      // Product Name
                      Text(
                        item["descriptor"]?["name"] ?? "Unnamed Product",
                        maxLines: 2,
                        overflow: TextOverflow.ellipsis,
                        style: const TextStyle(
                          fontSize: 14,
                          fontWeight: FontWeight.w600,
                          color: Colors.black87,
                        ),
                      ),

                      const SizedBox(height: 6),

                      // Price
                      Text(
                        "₹$priceValue / $unit",
                        style: const TextStyle(
                          fontSize: 14,
                          fontWeight: FontWeight.bold,
                          color: Colors.green,
                        ),
                      ),

                      const SizedBox(height: 4),

                      // Seller Info
                      Text(
                        "Seller: $providerName",
                        maxLines: 1,
                        overflow: TextOverflow.ellipsis,
                        style: const TextStyle(
                          fontSize: 12,
                          fontWeight: FontWeight.w500,
                          color: Colors.green,
                        ),
                      ),
                    ],
                  ),
                ),
              );
            },
          ),
        ),
      ],
    );
  }
}
